src/Temporalio/Activity/ActivityCancelReason.cs

```diff
@@ -0,0 +1,40 @@
+namespace Temporalio.Activity
+{
+    /// <summary>
+    /// Reason for why an activity may be cancelled.
+    /// </summary>
+    public enum ActivityCancelReason
+    {
+        /// <summary>
+        /// The activity has not been cancelled or was cancelled for an unknown reason.
+        /// </summary>
+        Unknown,
```

Comment:
> when will the reason be unknown?

Unknown when either given unknown from core or when the activity is never cancelled (but we cancel the token anyways after the activity returns so if they sent this context off somewhere else, this will remain unknown post activity completion though the cancel token will be done).